### DISCUSSION TITLE
screen flicker/flash when taking bitmap

### DISCUSSION THREAD

**Discussion by jpuerj** [iseahound/ImagePut#29] [Q&A] [answered]
Hi! I've seen this problem in the classic gdip library where the window flicker/blink/flash when taking the bitmap of the active window. This is happening for a specific program in my case. Is there a way to solve this? Thanks.

**Comment by iseahound** [comment]
Why not just use ImagePutFile({screenshot: "A"}) to take a screenshot instead of using the PrintWindow API as with ImagePutFile("A"). If it flickers its probably some accelerated window such as a game.

**Comment by jpuerj** [comment]
The problem is that I'm using ImagePutBuffer and then using the buffer to do ImageSearch. Is there a way to put it in a variable without passing it to a file?

**Comment by iseahound** [comment]
I've unsubscribed from my own repositories now to avoid receiving future notifications like this. Note that the issues tab is for issues with the code base and is not your personal help button. Please read the documentation. If that isn't possible ask in the autohotkey forum where someone can help with your question.

**Comment by iseahound** [comment, marked as answer]
The answer is to ask ImagePut to take a screenshot instead.

```
; Specify the input type as a screenshot to avoid window flicker.
; Replace "A" with the title of your window.
file := ImagePutFile({screenshot: "A"})  ; string to a filepath. 
buf := ImagePutBuffer({screenshot: "A"}) ; buffer object
```
PixelSearch Example:

    xy := buf.pixelsearch(0xFFFFFF)
    MsgBox xy[1], xy[2]
